### PR TITLE
More uses of ArgumentOutOfRangeException.ThrowIf* to replace custom error messages

### PR DESCRIPTION
**Pull request by halgab**
Follow-up to #8408

## Description

As mentioned in #8408, ParameterMustBeBetween, ParameterValueCannotBeInfinity and ParameterValueCannotBeNaN are good candidates for replacing custom ressources with inbox ones. This PR addresses those two cases, each in its own commit

## Regression

No

## Testing

CI

## Risk

Low. I manually searched and removed the error messages, and let CA1512 finish the work

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/wpf/pull/8489)

### CODE REVIEW

**Comment by lindexi** [comment]
Awesome, thank you @halgab for your contribution.

File: src/Microsoft.DotNet.Wpf/src/PresentationCore/Resources/ExceptionStringTable.txt [hunk 1]
```diff
@@ -15,7 +15,6 @@ UsesPerPixelOpacityIsObsolete=UsesPerPixelOpacity is obsolete and should not be
 ParameterMustBeGreaterThanZero=The parameter value must be greater than zero.
 ParameterCannotBeLessThan=The parameter value cannot be less than '{0}'.
 ParameterCannotBeGreaterThan=The parameter value cannot be greater than '{0}'.
-ParameterMustBeBetween=The parameter value must be between '{0}' and '{1}'.
```
[the remark this answers is not in the document]
Comment (rchauhan18) [reply src/Microsoft.DotNet.Wpf/src/PresentationCore/Resources/ExceptionStringTable.txt:18]: Yes, you can remove the lines here.

**Review by rchauhan18 — APPROVED** [review]

File: src/Microsoft.DotNet.Wpf/src/PresentationCore/Resources/ExceptionStringTable.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (halgab) [reply src/Microsoft.DotNet.Wpf/src/PresentationCore/Resources/ExceptionStringTable.txt:18]: done

**Review by rchauhan18 — APPROVED** [review]

**Comment by rchauhan18** [comment]
Thanks @halgab for the contribution!

**Pull request merged**